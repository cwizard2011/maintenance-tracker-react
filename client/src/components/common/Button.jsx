import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { value, requestId } = props;
  return (
    <div className="center">
      <Link to={`/request/${requestId}/edit`}>
        <a role="presentation" className="btn center">
          <ion-icon name="create" />
          {value}
        </a>
      </Link>
    </div>
  );
};
Button.propTypes = {
  value: PropTypes.string,
  requestId: PropTypes.string
};
export default Button;
