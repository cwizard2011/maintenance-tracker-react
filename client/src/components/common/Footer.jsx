import React from 'react';

const Footer = () => (
  <footer>
    <div>
        Maintenance Tracker by Adeoye Peter
    </div>
  </footer>
);

export default Footer;
