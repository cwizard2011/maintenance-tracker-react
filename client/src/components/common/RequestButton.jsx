import React from 'react';
import PropTypes from 'prop-types';

const RequestButton = (props) => {
  const {
    handleClick, iconName, actionName, buttonClass
  } = props;
  return (
    <span className="right-text">
      <button
        type="button"
        className={buttonClass}
        onClick={handleClick}
      >
        <ion-icon name={iconName} />
        {actionName}
      </button>
    </span>
  );
};
RequestButton.propTypes = {
  handleClick: PropTypes.func,
  iconName: PropTypes.string,
  actionName: PropTypes.string,
  buttonClass: PropTypes.string
};
export default RequestButton;
