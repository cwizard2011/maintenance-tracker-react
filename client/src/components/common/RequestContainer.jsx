import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const RequestContainer = (props) => {
  const {
    title, label, name, status, requestId
  } = props;
  return (
    <div className="curved-border">
      <h3>{title}</h3>
      <p>
      Status:
        {' '}
        <span className={label}>
          <ion-icon name="icon" />
          {status}
        </span>
        <span className="create">
          <Link to={`/request/${requestId}`} className="view-btn">
            <ion-icon name={name} />
            Details
          </Link>

        </span>

      </p>
    </div>
  );
};

RequestContainer.propTypes = {
  title: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string,
  status: PropTypes.string,
  requestId: PropTypes.string
};

export default RequestContainer;
