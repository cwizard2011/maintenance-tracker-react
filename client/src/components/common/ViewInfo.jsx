import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

export const ViewInfo = ({
  requestId, title, sender, email, details, date
}) => (
  <div>
    <div>
      <label htmlFor="id">Request Id</label>
      <p>{requestId}</p>
    </div>
    <div>
      <label htmlFor="sender">Sender</label>
      <p>{sender}</p>
    </div>
    <div>
      <label htmlFor="title">Title</label>
      <p>{title}</p>
    </div>
    <div>
      <label htmlFor="email">Email</label>
      <p>{email}</p>
    </div>
    <div>
      <label htmlFor="details">Details</label>
      <p>{details}</p>
    </div>
    <div>
      <label htmlFor="date">Date</label>
      <p>{moment(date).format('MMM D, YYYY')}</p>
    </div>
  </div>
);

export const LabelTag = ({ status, pClass, labelName }) => (
  <div>
    <label htmlFor="status">Status</label>
    <p>
      <span className={pClass}>
        <ion-icon name={labelName} />
        {status}
      </span>
    </p>
  </div>
);

ViewInfo.propTypes = {
  requestId: PropTypes.string,
  title: PropTypes.string,
  sender: PropTypes.string,
  email: PropTypes.string,
  details: PropTypes.string,
  date: PropTypes.string
};

LabelTag.propTypes = {
  status: PropTypes.string,
  pClass: PropTypes.string,
  labelName: PropTypes.string,
};
