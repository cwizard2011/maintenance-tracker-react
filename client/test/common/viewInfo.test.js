import React from 'react';
import { shallow } from 'enzyme';
import { ViewInfo, LabelTag } from '../../src/components/common/ViewInfo';

describe('View Info and Label Tag', () => {
  it('render View Info correctly', () => {
    const wrapper = shallow(<ViewInfo />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render Label Tag correctly', () => {
    const wrapper = shallow(<LabelTag />);
    expect(wrapper).toMatchSnapshot();
  });
});
