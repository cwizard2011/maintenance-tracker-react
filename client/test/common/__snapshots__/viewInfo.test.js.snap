// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Info and Label Tag render Label Tag correctly 1`] = `
<div>
  <label
    htmlFor="status"
  >
    Status
  </label>
  <p>
    <span>
      <ion-icon />
    </span>
  </p>
</div>
`;

exports[`View Info and Label Tag render View Info correctly 1`] = `
<div>
  <div>
    <label
      htmlFor="id"
    >
      Request Id
    </label>
    <p />
  </div>
  <div>
    <label
      htmlFor="sender"
    >
      Sender
    </label>
    <p />
  </div>
  <div>
    <label
      htmlFor="title"
    >
      Title
    </label>
    <p />
  </div>
  <div>
    <label
      htmlFor="email"
    >
      Email
    </label>
    <p />
  </div>
  <div>
    <label
      htmlFor="details"
    >
      Details
    </label>
    <p />
  </div>
  <div>
    <label
      htmlFor="date"
    >
      Date
    </label>
    <p>
      Nov 1, 2018
    </p>
  </div>
</div>
`;
